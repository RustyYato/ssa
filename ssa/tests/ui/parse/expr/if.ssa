if x {
    let a = d;
} else if y {
    let b = e;
} else {
    let c = f;
}

