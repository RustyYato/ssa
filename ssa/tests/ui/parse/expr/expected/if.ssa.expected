[
    id: 4
    kind:
        If:
            blocks: [
                cond:
                    id: 1
                    kind:
                        Ident:
                            id: 1
                            name: "x"
                block:
                    items: [
                        id: 1
                        kind:
                            Let:
                                binding:
                                    id: 2
                                    name: "a"
                                ty: null
                                value:
                                    id: 2
                                    kind:
                                        Ident:
                                            id: 3
                                            name: "d"
                    ],
                cond:
                    id: 3
                    kind:
                        Ident:
                            id: 4
                            name: "y"
                block:
                    items: [
                        id: 2
                        kind:
                            Let:
                                binding:
                                    id: 5
                                    name: "b"
                                ty: null
                                value:
                                    id: 4
                                    kind:
                                        Ident:
                                            id: 6
                                            name: "e"
                    ]
            ]
            default:
                items: [
                    id: 3
                    kind:
                        Let:
                            binding:
                                id: 7
                                name: "c"
                            ty: null
                            value:
                                id: 5
                                kind:
                                    Ident:
                                        id: 8
                                        name: "f"
                ]
]